src/pages/Product.jsx

```diff
@@ -0,0 +1,181 @@
+import React, { useEffect, useState } from 'react';
+import { useParams } from 'react-router-dom';
+
+import { useProduct } from '../utils/hooks/useProduct';
+
+//Components
+import { Button, Loading, NotFound } from '../components/common';
+
+//Helpers
+import numberWithCommas from '../utils/numberWithCommas.js'
+
+const Product = () => {
+	const { productId } = useParams()
+	const { data, isLoading } = useProduct(productId)
+
+	const [previewImg, setPreviewImg] = useState('')
+	const [descriptionExpand, setDescriptionExpand] = useState(false)
+	const [productNotFound, setProductNotFound] = useState(false)
+	const [productInfo, setProductInfo] = useState(null)
+  const [quantity, setQuantity] = useState(1)
+
+	useEffect(() => {
+		if (!isLoading && data.results_size > 0) {
+			setProductNotFound(true)
+			setProductInfo(data.results[0])
+      setPreviewImg(data?.results[0]?.data?.mainimage?.url)
+		}
+	}, [data, isLoading]);
+
+  const updateQuantity = (type) => {
+    const stock = productInfo?.data?.stock
+    type === 'plus' ? setQuantity(quantity + 1 <= stock ? quantity + 1 : quantity) 
+                    : setQuantity(quantity - 1 < 1 ? 1 : quantity - 1)
+  }
+
+return isLoading ? <Loading text='Loading product data...' />  
```

Comment:
I think this is a little bit difficult to read, maybe you could set up a formatter in order to help you with spaces and indentation, prettier is a very good one, you can install it on vscode